src/main/java/seedu/address/logic/parser/FindCustomerCommandParser.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.commons.util.StringUtil.splitToKeywordsList;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PHONE;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.FindCustomerCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.customer.Customer;
+import seedu.address.model.customer.predicates.CustomerAddressPredicate;
+import seedu.address.model.customer.predicates.CustomerEmailPredicate;
+import seedu.address.model.customer.predicates.CustomerNamePredicate;
+import seedu.address.model.customer.predicates.CustomerPhonePredicate;
+import seedu.address.model.util.predicate.CompositeFieldPredicateBuilder;
+
+/**
+ * Parses input arguments and creates a new FindCommand object
+ */
+public class FindCustomerCommandParser implements Parser<FindCustomerCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindCustomerCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+            ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS);
+
+        if (!areAnyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ADDRESS, PREFIX_PHONE, PREFIX_EMAIL)
+            || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCustomerCommand.MESSAGE_USAGE));
```

Comment:
Currently some funny things are happening to the customer list after we make the findcustomer command with an empty field following a valid prefix, for example, after inputting "findcustomer n/".

![Screen Shot 2021-03-22 at 7 22 06 PM](https://user-images.githubusercontent.com/77544764/111984285-26cef100-8b46-11eb-8106-69cb89e67c0a.png)

Perhaps we can consider using ParseException to handle the above case as well?